torch/_meta_registrations.py

```diff
@@ -6034,6 +6034,43 @@ def meta_local_scalar_dense(self: Tensor):
     raise RuntimeError("Tensor.item() cannot be called on meta tensors")
 
 
+@register_meta(aten._jagged_to_padded_dense_forward.default)
+def meta__jagged_to_padded_dense_forward(
+    values: Tensor,
+    offsets: List[Tensor],
+    max_lengths: List[int],
+    padding_value: float = 0.0,
+):
+    # only one jagged dim is supported for now
+    assert len(offsets) == 1
+    assert len(max_lengths) == 1
+
+    B = offsets[0].shape[0] - 1
+    S = max_lengths[0]
+    output_shape = (B, S, *values.shape[1:])
+    return values.new_empty(output_shape)
+
+
+@register_meta(aten._padded_dense_to_jagged_forward.default)
+def meta__padded_dense_to_jagged_forward(
+    padded: Tensor, offsets: List[Tensor], total_L: Optional[int] = None
+):
+    # only one jagged dim is supported for now
+    assert len(offsets) == 1
+
+    if not total_L:
+        assert isinstance(padded, torch._subclasses.FakeTensor)
+        shape_env = padded.fake_mode.shape_env
+        assert shape_env is not None
+        total_L = shape_env.create_unbacked_symint()
+        torch.fx.experimental.symbolic_shapes._constrain_range_for_size(
+            total_L, min=1, max=None
```

Comment:
I don't think this is necessarily true is it? You could technically have length of 0 for every batch I think

Reply:
IIRC I had to specifically set this to >= 1 to circumvent some error - let me see if I can get away with >= 0 though. I do agree that's conceptually what we want.

Reply:
Okay cool, I was wrong about the >= 1. For posterity, this is the assert that trips when the range is unconstrained:
https://github.com/pytorch/pytorch/blob/29081059b6534377698db4a1086d745c22f2f6eb/torch/fx/experimental/symbolic_shapes.py#L5328

Some testing shows that constraining to >= 0 works fine to avoid tripping this assert.